### PR TITLE
README: Add endpoint docs

### PR DESCRIPTION
**Pull request by dbrgn**
Hope everything is correct. I had to determine the information from the db schema and the source code.

Related: #87 and #88.

### CODE REVIEW

**Review by Makram95 — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -7,6 +7,15 @@ the API.
...
 ## API Endpoints
 
+### Authentication
+
+The API uses token based authentication. A sample request looks like this:
+
+    curl 'http://localhost:3000/api/measurements' -H "Authorization: Bearer 0123456789ABCDEF" -v
+
+All index and show resources are publicly available while writing to the API is permitted only
+consumers who provide a private api key.
```
Comment (schmijos) [inline README.md:17]: You need an API key for reading as well. We register it as public API key on the `ApiConsumer`.

File: README.md [hunk 2]
```diff
@@ -15,29 +24,110 @@ The following endpoints are RESTful and allow full CRUD operations.
 - `/api/measurements`
 - `/api/sponsors`
 
+You can post a measurement to the api the following way (use the private api key of ApiConsumer):
+
+    curl -X POST 'http://localhost:3000/api/measurements' \
+        -H "Content-Type: application/json" \
+        -H "Authorization: Bearer 0123456789ABCDEF" \
+        -d '{"sensor_id": 1, "temperature": 20.7, "custom_attributes": {"foo": "bar"}}'
```
Comment (schmijos) [inline README.md:32]: I'll remove the custom attributes right now. They're BS.

File: README.md [hunk 3]
```diff
@@ -15,29 +24,110 @@ The following endpoints are RESTful and allow full CRUD operations.
...
+| Field | Type | Description |
+| --- | --- | --- |
+| id | int | Sensor ID |
+| device_name | string | Sensor name |
+| caption | string | Caption describing the sensor |
+| created_at | uint | Unix timestamp (seconds) |
+| sponsor_id | uint? | Associated sponsor ID |
+| latitude | double? | Latitude (WGS84) |
+| longitude | double? | Longitude (WGS84) |
+| latest_temperature | double? | Latest temperature measurement |
```
Comment (schmijos) [inline README.md:54]: We're talking JSON. Those are all `number`.

**Review by schmijos — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (dbrgn) [reply README.md:17]: Ah, good point. Yes, this text is simply wrong (and I'd like to keep the API consumers that way for now).

File: README.md (hunk 3 above)
Comment (dbrgn) [reply README.md:54]: Good point.

**Pull request merged**